packages/beacon-state-transition/src/fast/util/cachedValidatorList.ts

```diff
@@ -13,11 +12,18 @@ export class CachedValidatorList<T extends phase0.Validator> implements List<T>
   tree: Tree;
   type: CompositeListType<List<T>>;
   persistent: MutableVector<T>;
+  createFlat: (t: TreeBacked<T>) => T;
 
-  constructor(type: CompositeListType<List<T>>, tree: Tree, persistent: MutableVector<T>) {
+  constructor(
+    type: CompositeListType<List<T>>,
+    tree: Tree,
+    persistent: MutableVector<T>,
+    createFlat: (t: TreeBacked<T>) => T
```

Comment:
I don't think we need this to be passed in, just use `createValidatorFlat` directly.